src/code/z_camera.c

```diff
@@ -526,282 +869,7416 @@ s32 Camera_GetWaterBoxBgCamSetting(Camera* camera, f32* waterY) {
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_camera/func_800CDE6C.s")
+        if ((Camera_BgCheckInfo(camera, &angledOffsetForwards, &camCollision) != 0) &&
+            Camera_CheckOOB(camera, &offsetForwards, &angledOffsetForwards)) {
+            // Always going to result in 28.0f?
+            distResult = OLib_Vec3fDistXZ(&offsetForwards, &camCollision.pos);
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_camera/func_800CE2B8.s")
+            // (-D_801B9E60 < (50.0f - distResult))
+            if ((D_801B9E5C != 2) || !(-D_801B9E60 >= (50.0f - distResult))) {
+                D_801B9E5C = 1;
+                distResult = 50.0f - distResult; // 22.0f
+                D_801B9E60 = distResult;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_camera/func_800CE5E0.s")
+                // if (distResult == 0.0f)
+                if (distResult == -distResult) {
+                    distResultAdj = 0.0f;
+                } else {
+                    distResultAdj = distResult;
+                }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_camera/func_800CE79C.s")
+                return distResultAdj;
+            }
+        }
+    } else {
+        // Turn and project 90 degrees right
+        yawForwardsOffset = focalActorPosRot.rot.y - DEG_TO_BINANG(90);
+
+        angledOffsetForwards.x = (Math_SinS(yawForwardsOffset) * 50.0f) + offsetForwards.x;
+        angledOffsetForwards.y = offsetForwards.y;
+        angledOffsetForwards.z = (Math_CosS(yawForwardsOffset) * 50.0f) + offsetForwards.z;
+
+        if ((Camera_BgCheckInfo(camera, &angledOffsetForwards, &camCollision) != 0) &&
+            Camera_CheckOOB(camera, &offsetForwards, &angledOffsetForwards)) {
+            // Always going to result in 28.0f?
+            distResult = OLib_Vec3fDistXZ(&offsetForwards, &camCollision.pos);
+
+            if ((D_801B9E5C != 1) || !(D_801B9E60 >= -(distResult - 50.0f))) {
+                D_801B9E5C = 2;
+                distResult = distResult - 50.0f;
+                D_801B9E60 = distResult;
+                if (distResult == -distResult) {
+                    distResultAdj = 0.0f;
+                } else {
+                    distResultAdj = distResult;
+                }
+                return distResultAdj;
+            }
+        }
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_camera/func_800CE930.s")
+    if (D_801B9E5C != 0) {
+        distResult = D_801B9E60;
+    } else {
+        distResult = 0.0f;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_camera/func_800CEAD8.s")
+    D_801B9E5C = 0;
+    D_801B9E60 = 0.0f;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_camera/func_800CEC38.s")
+    return distResult;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_camera/func_800CED90.s")
+/**
+ * Calculates a new Up vector from the pitch, yaw, roll
+ */
+Vec3f* Camera_CalcUpFromPitchYawRoll(Vec3f* viewUp, s16 pitch, s16 yaw, s16 roll) {
```

Comment:
Maybe shorten the name to CalcUpVector.
Pitch, yaw, and roll are named parameters to the function. They don't need to be repeated.